pets/etcd/etcd-petset.yaml

```diff
@@ -0,0 +1,121 @@
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+  name: etcd
+  labels:
+    app: etcd
+spec:
+  ports:
+  - port: 2380
+    name: peer
+  - port: 2379
+    name: client
+  clusterIP: None
+  selector:
+    app: etcd
+---
+apiVersion: apps/v1alpha1
+kind: PetSet
+metadata:
+  name: etcd
+spec:
+  serviceName: "etcd"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: etcd
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      containers:
+      - name: etcd
+        image: gcr.io/google_containers/etcd-amd64:2.2.5
+        ports:
+        - containerPort: 2380
+          name: peer
+        - containerPort: 2379
+          name: client
+        env:
+        - name: CLUSTER_SIZE
```

Comment:
Reading the ``peer-discovery`` and the etcd docs, I need at least ``_etcd-server._tcp.example.com`` srv record for each pet. Anyway, the DNS will only save some time at bootstrapping. It will not help with scaling.

In order to add a new member I need to distinguish between bootstrapping phase and after-bootstrapping one. At the same time it is not recommended to change to size from 3 to 5. It should be done one by one. At the same time if you decrease the size, random pet gets killed. I don't think there is a way for a pod to do "the last wish". At the same time if the pet is the etcd leader, it will take more time to elect new one.

Atm, member can be added/remove only via runtime configuration [1] as:

```sh
$ etcdctl member add infra3 http://10.0.1.13:2380
$ etcdctl member remove a8266ecf031671f3
```

[1] https://coreos.com/etcd/docs/latest/runtime-configuration.html